src/passes/Asyncify.cpp

```diff
@@ -1497,6 +1510,8 @@ struct Asyncify : public Pass {
   void run(Module* module) override {
     auto& options = getPassOptions();
     bool optimize = options.optimizeLevel > 0;
+    is64 = module->memories.size() && module->memories[0]->is64();
+    pointerType = is64 ? Type::i64 : Type::i32;
```

Comment:
```suggestion
    pointerType = module->memories[0]->indexType;
```

Reply:
IIRC the problem here is that module->memories might be empty.   I guess asyncify should probably error out in that case?    But that seems like a bigger change.

Reply:
Oh, right, sorry, my suggested diff here isn't right then. Still, using `indexType` might simplify things.